### DISCUSSION TITLE
Vanilla javascript port

### DISCUSSION THREAD

**Discussion by michaelessiet** [mdx-editor/editor#508] [General]
Will there ever be a vanilla js port? If so I would like to assist in its development. If there was already one but is no longer maintained I would like to pick it up. If there will never be a port then that is such a shame, as this would be a game changing library if is were available to the javascript ecosystem as a whole and not just the React ecosystem.

If anyone is interested in starting a vanilla js port then I would like to assist you in any way I can whether as a maintainer or sponsor if time does not permit me to assist you in maintenance.

**Comment by petyosi** [comment]
There's no vanilla version and I don't plan on building one unless someone decides to sponsor several months of effort on my side. 

The core of the editor is framework agnostic, but rebuilding the UI (dialogs, toolbars, editors etc) without the Radix primitivies would be a massive effort, not to mention the complications it will bring to the architecture.

**Reply by michaelessiet** [reply]
Ok, how much would the sponsorship need to be per month and how long you reckon it would take? For now I plan on creating an app using this package with Next.js since a vanilla version doesn't exist. If the app later ends up gaining revenue I wouldn't mind directing some of that towards sponsoring the creation of a fully vanilla js version of this package

**Reply by petyosi** [reply]
So far, you're the first one to express an interest into such development. This means that you will have to subsidize the entire development.  If you're interested, you can reach me at petyo@virtuoso.dev.
